### PR TITLE
feat(planning_test_utils): add planning interface test manager for some obstacle modules

### PR DESCRIPTION
**Pull request by kyoichi-sugahara**
### background
There are many cases where planning nodes die when they input an empty trajectory, a trajectory with only one point, or a trajectory where the points are duplicated.

### purpose
To prevent these errors in advance, added tests to each node(some obstacle modules in this PR) to ensure that they do not crash when irregular trajectories is given.  Confirmed that the process works properly by providing a normal trajectory as input before giving an irregular trajectory and outputting the necessary information for subsequent processing.

### diagram
Here is an example of obstacle_cruise_planner_node and a diagram showing the test process.
The confirmation of normal operation is verified following the diagram and flow below.
Screenshot from 2023-03-14 17-53-41

Verification of behavior for irregular inputs is confirmed by following the diagram and flow below.
Screenshot from 2023-03-14 17-52-11

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by TakaHoribe — COMMENTED** [review]
@kyoichi-sugahara Thank you for your work. 
Overall, LGTM. The name `manager` sounds too much for this function. Could it be `utils` or something like that?

**Comment by kyoichi-sugahara** [comment]
@TakaHoribe 
Thank you!
I changed package name and I have made the necessary changes to adapt to other modules and confirmed that the tests have passed.

**Review by TakaHoribe — APPROVED** [review]

File: planning/motion_velocity_smoother/CMakeLists.txt [hunk 1]
```diff
@@ -47,11 +47,17 @@ rclcpp_components_register_node(motion_velocity_smoother_node
...
 if(BUILD_TESTING)
   ament_add_ros_isolated_gmock(test_smoother_functions
-    test/test_smoother_functions.cpp
+  test/test_smoother_functions.cpp
   )
   target_link_libraries(test_smoother_functions
-    smoother
+  smoother
   )
+  # ament_add_ros_isolated_gtest(test_${PROJECT_NAME}
```
Comment (satoshi-ota) [inline planning/motion_velocity_smoother/CMakeLists.txt:55]: May I ask you why it comment out only motion velocity smoother's test?

**Review by satoshi-ota — APPROVED** [review]

**Review by takayuki5168 — APPROVED** [review]

File: planning/motion_velocity_smoother/CMakeLists.txt (hunk 1 above)
Comment (kyoichi-sugahara) [reply planning/motion_velocity_smoother/CMakeLists.txt:55]: Because implemented test will fail with `motion_velocity_smoother` module...

**Pull request merged**